### DISCUSSION TITLE
Zoom does not work

### DISCUSSION THREAD

**Discussion by Pewibe** [jomjol/AI-on-the-edge-device#3006] [Q&A] [answered]
I like to enlarge part of my reference image (see attached screenshot). Therefore I used config params like zoom=true, zoom mode=1, zoom offset x=300, offset y=200 but I receive always the same picture. May I misunderstand something regarding zoom function? How do I have to config that my water meter circle is enlarged to maximal size?
![Clipboard01](https://github.com/jomjol/AI-on-the-edge-device/assets/42893315/7ee48d05-412c-445f-86d8-03d4da14ae8f)

**Comment by SybexX** [comment]
in zoom_mode = 0 should be max_offset_x = 160 and max_offset_y = 120

![315541676-4b6deced-3c83-4126-b5f4-4333f02ed3ac](https://github.com/jomjol/AI-on-the-edge-device/assets/587201/a86f00e8-9b32-4589-bb2c-0cb9a888ebbd)
![zoom_mode1](https://github.com/jomjol/AI-on-the-edge-device/assets/587201/f8f0b002-03d0-4ebc-be71-438a49fa8eea)

**Reply by Pewibe** [reply]
Thanks for the help but even with x/y = 120/160 Ihave seen no difference in my reference image. 
1. Is your answer just theory or checked in pactice?
2. What is the black rectangle in the drawings?
3. What is your suggestion for the zoom params to enlarge my centric watermeter?

**Reply by SybexX** [reply]
to 1.
yes, I tested it and it works for me, but if I specify the offset too high with zoom_mode = 0, the ESP reboots
to 2.
These resolutions are used when zooming and the desired area is cut out of these image
to 3.
try whether zoom_mode = 0 is enough, if it is not enough, just reducing the distance between the watermeter and ESP32 cam helps
or https://jomjol.github.io/AI-on-the-edge-device-docs/Installation/ 1.1.4 Housing

![Zoom](https://github.com/jomjol/AI-on-the-edge-device/assets/587201/6a9b8d6c-7186-4f18-9511-5847f0af1fda)

**Reply by Pewibe** [reply]
Sorry for my repeating questions. I can't follow you and your answer. My image is 640x480 and the cyrcle line of my water meter is always at the same pixels whether I use this param or that.
Just for clarification: I use version 15.6.0 because a lot of people and I have trouble with the actual 15.7.0. I select menu item "Settings/Alignment/Reference Image and Camera Settings", click once the button "Create New Reference ...", check "Show Expert Params", change several zoom params, finally click button "update Reference Image", and watch the updated Reference Image below which NEVER changes. To check the opposite I open the next water tap for a liter and the analog wheel is moved when I update the reference image again.
Do I do something other compared to your usage? Any last hint before I give up and perform your suggested "manual zoom"?

P.S.: My last brainstorming reminds me that I downgraded from 15.7.0 to 15.6.0 the ESP only, not the SD card. I thought that the web interface can't be an error source. I you say that everything I did is OK, I shall downgrade the SD card, too.

**Reply by SybexX** [reply, marked as answer]
Version 15.6.0 does not have a zoom function

**Comment by caco3** [comment]
Instead of zooming and loosing a lot of the cam resolution, can't you move the cam closer? That should be the first thing to do before using digital zoom.

**Reply by Pewibe** [reply]
Yes, I did it. Sometimes I like to understand features even if I do not use them.
